src/fs/rock/RockRebuild.cc

```diff
@@ -694,19 +697,41 @@ Rock::Rebuild::swanSong()
     storeRebuildComplete(&counts);
 }
 
+void
+Rock::Rebuild::callException(const std::exception &e)
+{
+    ErrorDetail::Pointer errorDetail;
+    if (const auto tex = dynamic_cast<const TextException*>(&e))
+        errorDetail = new ExceptionErrorDetail(tex->id());
+    else
+        errorDetail = new ExceptionErrorDetail(Here().id());
+    failure(ToSBuf(e.what(),
+        Debug::Extra, errorDetail).c_str());
+    // fatal failure() substitutes default AsyncJob::callException(e) handling
```

Comment:
callExcpetion() is a custom exception handler that prevents Rebuild job exceptions from being silently ignored _and_ that kills the process. We may make exceptions non-fatal later, but, hopefully, not in this PR; the PR description already discusses why.

AFAICT, we just need to (effectively) re-throw the caught exception here. I hope the suggested code is the right way to do that:

```suggestion
    // For now, treat all Build exceptions as fatal errors rather than letting
    // the default callException() implementation to silently stop this job.
    thow;
```

Reply:
Done.